torch/csrc/autograd/function.h

```diff
@@ -104,6 +104,7 @@ struct TORCH_API Node : std::enable_shared_from_this<Node> {
       uint64_t sequence_nr,
       edge_list&& next_edges = edge_list())
       : sequence_nr_(sequence_nr),
+      thread_id_(at::RecordFunction::currentThreadId()),
```

Comment:
It's not something I like to see for sure. Though I never actually did a full benchmark of the impact on memory and speed.

Do you actually always need this thread id or just when the profiler is running?
If so, I think a nice way to guard this to only run when the profiler runs and store it in the metadata struct that we also use for anomaly mode to store this (you can make the python dict initialization lazy to improve perf when you just store and read the thread_id).


> I'm not sure how in general sequence_nrs work - are they thread local?

Yes these are just thread local variables (not propagated by TLSState)

Reply:
sequence numbers are thread local, that's why i needed to add original fwd thread id, otherwise the fwd-backward matching can be incorrect

Reply:
we then should either set or not set both of them, because seq. number without thread is not very useful;
if extra int64 is too much I'd better go with packing though - less runtime checks

Reply:
The autograd engine make heavy use of the seq number (that does not require the thread). So we have to set this one.